toolkit/tools/internal/targetos/targetos.go

```diff
@@ -0,0 +1,69 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+package targetos
+
+import (
+	"fmt"
+	"path/filepath"
+	"strings"
+
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+)
+
+type TargetOs string
+
+const (
+	TargetOsAzureLinux2 TargetOs = "azl2"
+	TargetOsAzureLinux3 TargetOs = "azl3"
+)
+
+func GetInstalledTargetOs(rootfs string) (TargetOs, error) {
+	osReleaseLines, err := file.ReadLines(filepath.Join(rootfs, "etc/os-release"))
+	if err != nil {
+		return "", fmt.Errorf("failed to read /etc/os-release file:\n%w", err)
+	}
+
+	fields := make(map[string]string)
+
+	for _, line := range osReleaseLines {
+
+		split := strings.SplitN(line, "=", 2)
+		if len(split) < 2 {
+			continue
+		}
+		name := split[0]
+		value := split[1]
+
+		value = strings.TrimPrefix(value, "\"")
+		value = strings.TrimSuffix(value, "\"")
+
+		fields[name] = value
+	}
```

Comment:
Just an idea: this looks like a great candidate to be extracted to a separate helper function for reading `/etc/os-release` into a map.

Reply:
You could also use this Go module, although it's not super fresh: https://github.com/acobaugh/osrelease.

There's also this one: https://pkg.go.dev/github.com/quay/claircore/osrelease. Seems fresher (last commit a week ago): https://github.com/quay/claircore/tree/main/osrelease.

Reply:
I split the `os-release` parsing into its own package.

I also figured I'd used the robust grub/Bash parser I've already written.